vignettes/funbiogeo.Rmd

```diff
@@ -20,405 +21,296 @@ knitr::opts_chunk$set(
-`fb_format_*()` to create these specific datasets. This implies that your data 
-are structured into a long table, i.e. one row corresponds to the 
-occurrence/abundance/coverage of one species at one site. This also implies that
-some variables are repeated across the table (e.g. sites coordinates, species 
-traits).
+`funbiogeo` provides many functions to display the data to help the use select
+specific traits, species, and/or sites. We are going to detail some of them
+in this section. We call them *diagnostic plots* because they help us
+to have an overview of our dataset **prior** to the analyses.
 
-Let's take an example
+### Trait completeness per species
 
-```{r 'load-raw-dataset'}
-## Path to example raw dataset ----
-filename <- system.file("extdata", "raw_mammals_data.csv", package = "funbiogeo")
+A first way to visualize our dataset is to look at the proportion of species
+with non-missing traits using the `fb_plot_number_species_by_trait()` function.
+It takes the species by trait data.frame as input.
 
-## Read CSV file ----
-all_data <- read.csv(filename)
+```{r fig-plot-sp-by-trait}
+fb_plot_number_species_by_trait(species_traits)
 ```
 
-```{r 'preview-raw-dataset', echo = FALSE}
-knitr::kable(
-  head(all_data, 10), 
-  caption   = "Long table example", 
-  align     = c("c", "r", "r", "l", "c", "r", "r", "r", "r", "r", "r")
-)
-```
-
-<br />
-
-**Formatting site x locations data**
+This plot shows us the number of species (along the x-axis) in function of 
+the trait name (along the y-axis). The number of concerned species is shown at
+the bottom of the plot while the corresponding proportion of species
+(compared to all the species included in the trait dataset) is indicated as
+a secondary y-axis at the top. The proportion of species concerned is shown at
+the right of each point. For example, in our example dataset, 82.6% species have
+a non-NA adult body mass.
 
-The function `fb_format_site_locations()` extracts sites coordinates from this
-long table to create the site x locations dataset. Note that one site must have
-one unique longitude x latitude value.
+The function also include a way to provide a target proportion of species as
+the second argument. It will display the proportion as a dark red dashed line.
 
-```{r 'format-sites-locs'}
-## Format site x locations data ----
-site_locations <- fb_format_site_locations(input_data =  all_data, 
-                                           site       = "site", 
-                                           longitude  = "longitude", 
-                                           latitude   = "latitude",
-                                           na_rm      = FALSE)
+For example, if we want to visualize which traits cover more than 75%
+of the species:
 
-## Preview ----
-head(site_locations)
+```{r fig-plot-sp-by-trait-prop}
+fb_plot_number_species_by_trait(species_traits, 0.75)
 ```
 
-<br />
+The top number shows the corresponding number of species.
```

Comment:
This number should be rounded (no decimal) as it corresponds to a number of species.